api/mixins.py

```diff
@@ -0,0 +1,42 @@
+from rest_framework import status
+from rest_framework.decorators import action
+from rest_framework.response import Response
+
+from api.serializers import UserSerializer
+from likes import services
+from likes.models import LikeDislike
+
+
+class LikedMixin:
+    @action(methods=['POST'], detail=True, url_path='vote/(?P<vote_type>\\w+)')
```

Comment:
issue: так доступно только админу
```suggestion
    @action(methods=['POST'], detail=True, url_path='vote/(?P<vote_type>\\w+)', permission_classes=(IsAuthenticated,),)
```